### ISSUE TITLE
Native R on Travis takes longer than the script

### ISSUE THREAD

**Issue by robertzk** [craigcitro/r-travis#147]
![screen shot 2015-02-26 at 10 55 35 am](https://cloud.githubusercontent.com/assets/1638492/6396902/f243c42e-bda6-11e4-835d-6a2683d8f951.png)

On my packages, building using the new native R support takes 3-4x as long as using the old method. Is it possible to incorporate a .travis.yml flag to install just base R to speed up this time?

**Comment by eddelbuettel** [comment]
Same here on a first test with one of the the most minimal CRAN packages I have.    I had tuned my .travis.yml carefully to be as lightweight as possible; the way things stand right now the new default seems useful for newbs but I guess I will cook my own (making things like texlive, pandoc, automated addition of two apt-get repos etc pp all optional.)

**Comment by eddelbuettel** [comment]
In a nutshell, what happened was:

* while we all worked on r-travis, we made sure certain "API" access points were optional
* now latex, devtools, ... are forced in unconditionally

and in sum everything seems to take _several minutes_ minutes longer.

**Comment by craigcitro** [comment]
yes, this is by design: we unconditionally install latex and pandoc, and no longer set the `R CMD check` options to skip vignettes.

but don't worry, this is all part of a plan: now that we're merged into travis, we're looking at getting ourselves a base image with all of these bits pre-installed. then we'll have fast builds *with* all the various tools available.

and if you're wondering why i didn't leave it all optional to start: i didn't want to introduce a whole collection of configuration options that were slated to go away from the start, making deprecating them later potentially more complicated.

**Comment by craigcitro** [comment]
(i'm going to leave this open for other folks who come looking.)

**Comment by ehrlinger** [comment]
My package compiles a vignette with LaTeX, but still fails on Travis because of missing latex packages. Is there an option to either install the missing `xcolor` or `booktabs` packages in LaTeX? Or alternatively, skip the vignettes?

**Comment by eddelbuettel** [comment]
Sure:
+ I have been skipping vignette builds since the start as it takes so long to set it up
+ If you want to build with vignettes and need extra packages, use the apt-get hook to install those.

**Comment by ehrlinger** [comment]
Got it working again... with:
```
apt_packages:
  - texlive-full
```
May have been brute force, but it succeeds.
